src/Highlighter.ts

```diff
@@ -127,6 +133,28 @@ export default class Highlighter {
     }
   }
 
+  private onFocusHandler = (type: 'in' | 'out') => (ev: Event): void => {
+    if (
+      (type === 'in' && !this.options.onFocusIn)
+      || (type === 'out' && !this.options.onFocusOut)
+    ) {
+      return;
+    }
+
+    const highlightId = isHtmlElement(ev.target) && ev.target.hasAttribute(DATA_SCREEN_READERS_ATTR)
+      ? ev.target.getAttribute(DATA_ID_ATTR)
+      : null;
+    const highlight = highlightId ? this.getHighlight(highlightId) : null;
+
+    if (type === 'in' && highlight) {
+      this.options.onFocusIn!(highlight);
```

Comment:
you shouldn't need these `!`. limit your conditionals to a section at the top where you extract a `handler` variable, then return if the handler is undefined, the rest of the logic can work as normal without knowledge of whether its out or in and will know the handler is defined